Gulpfile.js

```diff
@@ -265,6 +278,124 @@ gulp.task('test-client-travis', ['build'], function () {
 
 gulp.task('travis', [process.env.GULP_TASK || '']);
 
+//Documentation
+gulp.task('lint-docs', function () {
+    function lintFiles (files) {
+        return new Promise(function (resolve, reject) {
+            var config = {
+                'files':  files,
+                'config': require('./.markdownlint.json')
+            };
+
+            markdownlint(config, function (err, result) {
+                var lintErr = err || result && result.toString();
+
+                if (lintErr)
+                    reject(lintErr);
+                else
+                    resolve();
+            });
+        });
+    }
+
+    return globby('./docs/articles/**/*.md').then(lintFiles);
+});
+
+gulp.task('clear-site', ['lint-docs'], function (cb) {
+    del('./site/**', function (e) {
```

Comment:
No need for relative paths in gulpfile and for globs here as well. Just `del('stie', cb); See `clean` implementation above. Also, maybe we should rename it to `clean-site` for consistency. Also, some tasks use term `website` others just `site`, dir is named `site` as well. We should end up we one common term. I prefer `website`